modules/core/src/main/java/org/apache/synapse/config/xml/MessageStoreMediatorFactory.java

```diff
@@ -53,7 +54,21 @@ protected Mediator createSpecificMediator(OMElement elem, Properties properties)
         OMAttribute messageStoreNameAtt = elem.getAttribute(ATT_MESSAGE_STORE);
 
         if(messageStoreNameAtt != null) {
-            messageStoreMediator.setMessageStoreName(messageStoreNameAtt.getAttributeValue());
+
+            if(checkForBraces(messageStoreNameAtt)){
+                String newpath = removeBraces(messageStoreNameAtt);
+                try {
+                    messageStoreMediator.setMessageStoreExp(SynapsePathFactory.getSynapsePath(elem, newpath));
+                } catch (JaxenException e) {
+                    String msg = "Invalid XPath expression for attribute 'messageStore' : " +
+                            messageStoreNameAtt.getAttributeValue();
+                    log.error(msg);
+                    throw new SynapseException(msg);
```

Comment:
Add the exception object to the SynapseException. Otherwise, we might miss some information.

Reply:
done